### ISSUE TITLE
By default, use onlineticket, then fallback to onlinekey.

### ISSUE THREAD

**Issue by llakssz** [llakssz/FunKiiU#22]
At the moment you need to specify -onlineticket.
-onlineticket will be a legit ticket so it is the best choice to always use, without a doubt.
but if an onlineticket does not exist, instead, use the key (if that exists in the json instead)

If nothing exists, of course, give an error.

when using '-all' (now it is -region USA, -region EUR USA), default to onlinetickets, and fallback like above?

For example, should we type:
FunKiiU.py -region USA
or
FunKiiU.py -region USA -onlinetickets

and, if typing online tickets, it would only download content that has an online ticket, missing out on all stuff like dlc.

soon we will have easy signature patches, and although nice, a legit ticket will not be so important.

When using -onlinetickets, assume the user only wants the legit ticket, and if it doesn't exist, quit?
Or wait for user input, 'Do you want to get the key instead?' -- I don't want this program to be interactive...

Difficult decisions, thankfully there cannot be any upset customers ;)

**Comment by life-elevated** [comment]
I agree that a tool like this should not be interactive. That's what guis are for. When using onlinetickets, I would think that grabbing content with a key instead of a ticket if there isn't one wouldn't be the expected behavior. The user would need to be notified that it couldn't find an online ticket and a key was used to create the ticket. Otherwise they might assume they have a title with a legit ticket since that was the expected output. They will try to install without sigpatches and blow up the forums with questions. At least for now until sigpatched fws are readily usable.

**Comment by llakssz** [comment]
Yes all good points, thank you :)

Maybe appending a '***' or 'legit' to the folder name, if a legit ticket was used?
Since, at the very least it would be easy to forget.

(Note that it is 'easy' to check if the ticket is generated or not, I have some custom hex text in the tickets I make... poor Isabella)

**Comment by life-elevated** [comment]
I think that could work. As long as it was obvious which ones were which.
